pytest/tests/mocknet/helpers/neard_runner.py

```diff
@@ -461,25 +476,57 @@ def do_stop(self):
                 self.set_state(TestState.STOPPED)
                 self.save_data()
 
-    def do_reset(self):
+    def do_reset(self, backup_id=None):
         with self.lock:
             state = self.get_state()
             logging.info(f"do_reset {state}")
+            if state != TestState.RUNNING and state != TestState.STOPPED:
+                raise jsonrpc.exceptions.JSONRPCDispatchException(
+                    code=-32600,
+                    message='Cannot reset data dir as test state is not ready')
+
+            try:
+                backups = self.data['backups']
+            except KeyError:
+                backups = {}
+            if backup_id is not None and backup_id != 'start' and backup_id not in backups:
+                raise jsonrpc.exceptions.JSONRPCDispatchException(
+                    code=-32600, message=f'backup ID {backup_id} not known')
+
+            if backup_id is None or backup_id == 'start':
+                path = self.data['binaries'][0]['system_path']
```

Comment:
ah actually i did include that, but i wanted to add this special case here so that this will work if you start `neard_runner.py` that includes this PR on a directory that was made by an older version, so it will "just work" in that case. same kind of thing behind `backup_id` being an optional arg of this RPC method, so that an older `mirror.py` will work with a newer `neard_runner.py` running on the mocknet nodes